src/compiler/args.ml

```diff
@@ -162,7 +162,7 @@ let parse_args com =
 				let all,max_length = Define.get_user_documentation_list com.user_defines in
 				let all = List.map (fun (n,doc) -> Printf.sprintf " %-*s: %s" max_length n (limit_string doc (max_length + 3))) all in
 				List.iter (fun msg -> com.print (msg ^ "\n")) all;
-				exit 0
+				raise Abort
```

Comment:
I'm fine with making the change but this really doesn't look right. A nice `exit 0` (= everything is fine) should not be equivalent to a `raise Abort` (= I give up), so that's something we should clean up properly.

Reply:
I'm not sure what the actual difference is, but `raise Abort` does the same thing (stop compilation with `0` as exit code).

Problem was that `exit 0` meant stopping haxe server when we do `haxe --connect X --help-*` which is pretty bad.

Reply:
Yeah this is just a naming thing that throws me off, because to me aborting something suggests that it failed. But I don't really know what else to call it either, so never mind!

Reply:
Yeah at first I tried `Exit` which pretty much did what we'd expect `Abort` to do, then tried `Abort` and it did exactly what I was after x)

Reply:
`Exit` is a control-flow error like `Not_found` that shouldn't flop up the call stack too much. Not a great name but that's an OCaml stdlib thing.

I'll rename `Abort` to `StopCompilation` (after merging this to avoid conflicts) and add some general exception documentation.